conn.go

```diff
@@ -108,10 +113,133 @@ func (c *Conn) RemoteAddr() net.Addr {
 	return c.conn.RemoteAddr()
 }
 
-// Write takes a frisbee.Message and some (optional) accompanying content, and queues it up to send asynchronously.
+// Write takes a byte slice and sends a BUFFER message
+func (c *Conn) Write(p []byte) (int, error) {
+
+	var encodedMessage [protocol.MessageV0Size]byte
+
+	binary.BigEndian.PutUint16(encodedMessage[protocol.VersionV0Offset:protocol.VersionV0Offset+protocol.VersionV0Size], protocol.Version0)
+	binary.BigEndian.PutUint32(encodedMessage[protocol.OperationV0Offset:protocol.OperationV0Offset+protocol.OperationV0Size], BUFFER)
+	binary.BigEndian.PutUint64(encodedMessage[protocol.ContentLengthV0Offset:protocol.ContentLengthV0Offset+protocol.ContentLengthV0Size], uint64(len(p)))
+
+	c.Lock()
+	if c.state.Load() != CONNECTED {
+		c.Unlock()
+		return 0, c.Error()
+	}
+
+	_, err := c.writer.Write(encodedMessage[:])
```

Comment:
why are you copying the slice here?

Reply:
Because it's not a slice - it's a fixed size array, I need to convert it to a slice for the `Write` function to accept it